test/logflare_web/controllers/plugs/partner_authentication_test.exs

```diff
@@ -0,0 +1,46 @@
+defmodule LogflareWeb.Plugs.PartnerAuthenticationTest do
```

Comment:
Should use the VerifyApiAccess plug that is also worked on in #1392  and discussed in #1396 , we can extend it to have scope verification, so partner routes would have scope checking.